os/include/tinyara/bluetooth/bt_core.h

```diff
@@ -155,7 +155,7 @@ struct bt_eir_s {
 
 static inline int bt_addr_to_str(FAR const bt_addr_t *addr, FAR char *str, size_t len)
 {
-	return snprintf(str, len, "%2.2X:%2.2X:%2.2X:%2.2X:%2.2X:%2.2X", addr->val[5], addr->val[4], addr->val[3], addr->val[2], addr->val[1], addr->val[0]);
+	return snprintf(str, len, "%02X:%02X:%02X:%02X:%02X:%02X", addr->val[0], addr->val[1], addr->val[2], addr->val[3], addr->val[4], addr->val[0]);
```

Comment:
Please change addr->val[0]); to addr->val[5);

Reply:
Address is converted like below on capi layer. so, I changed this code sams as capi.

431 int _bt_convert_address_to_string(char **addr_str, unsigned char *addr_hex)
432 {
433     char address[18] = { 0, };
434 
435     BT_CHECK_INPUT_PARAMETER(addr_hex);
436 
437     snprintf(address, 18, "%02X:%02X:%02X:%02X:%02X:%02X", addr_hex[0], addr_hex[1], addr_hex[2], addr_hex[3], addr_hex[4], addr_    hex[5]);
438     *addr_str = strdup(address);